release_notes/ocp-4-0-release-notes.adoc

```diff
@@ -0,0 +1,778 @@
+version 4.1 and later.
+
+The Node Tuning Operator helps you manage node-level tuning by orchestrating the
+tuned daemon. The majority of high-performance applications require some level
+of kernel tuning. The Node Tuning Operator provides a unified management
+interface to users of node-level sysctls and more flexibility to add custom
+tuning, which is currently a Technology Preview feature, specified by user
+needs. The Operator manages the containerized tuned daemon for OpenShift
+Container Platform as a Kubernetes DaemonSet. It ensures the custom tuning
+specification is passed to all containerized tuned daemons running in the
+cluster in the format that the daemons understand. The daemons run on all nodes
+in the cluster, one per node.
+
+[[ocp-4-1-metrics-and-logging]]
+=== Metrics and logging
+
+[[ocp-4-1-metering]]
+=== Metering
+
+[[ocp-4-1-metering-and-chargeback]]
+==== Metering and chargeback (Technology Preview)
+
+Metering and chargeback is a Technology Preview feature in {product-title} 4.1.
+
+The Operator Metering portion of the Operator framework gives administrators
+insight into usage for specific Operators. Usage metrics are unique to each
+Operator; metering watches custom resources and collects data, and performs
+calculations and reports on that data back to the administrator.
+
+[[ocp-4-1-cluster-monitoring]]
+==== Cluster monitoring
+
+{product-title} 4.1 provides a single, unified, and native experience for
+cluster administrators to manage alerts and start troubleshooting. It also
+provides a streamlined experience to set up app monitoring with Prometheus for
+tenants through the OperatorHub.
+
+[[ocp-4-1-autoscake-pods-horizontally-based-on-prometheus-metrics]]
+==== Autoscale pods horizontally based on any metrics in Prometheus (Technology Preview)
+
+This feature, currently in Technology Preview, enables you to configure
+horizontal pod autoscaling (HPA) based on any metric provided by Prometheus.
+Autoscale based on any cluster-level metrics {product-title} provides out of the
+box and autoscale based on any app metrics.
+
+Limitations:
+
+The adapter only connects to a single Prometheus instance (or a set of
+load-balanced replicas, using Kubernetes services). Manually deploying adapter
+and configuring it to use Prometheus (that includes Prometheus instance shipped
+in OpenShift) For using cluster-metrics, a cluster administrator needs to
+whitelist metrics that can be used to configure HPA.
+
+[[ocp-4-1-developer-experience]]
+=== Developer experience
+
+[[ocp-4-1-code-ready-containers]]
+==== Code ready containers
+
+A local desktop instance of {product-title} 4.1 replaces the functions of `oc
+cluster` commands, Minishift, and CDK. {product-title} 4.1 focuses on ease of
+access and native experience, with a native installer on macOS and Microsoft
+Windows, native hypervisor support, and tray icon integration.
+
+[[ocp-4-1-registry]]
+=== Registry
```

Comment:
should something be said about the fact that the registry is managed by an operator instead of oc adm registry, here?

/cc @dmage @adambkaplan

Reply:
Should we write the same sentence about each component? Because today everything is managed by operators. But we can mention that the registry has reasonable defaults and additional tuning should be done via images.config.openshift.io and configs.imageregistry.operator.openshift.io.

Reply:
it's true of everything in some sense but in the case of the registry i think it's a more notable change because of the switch of `oc adm registry`.  I don't think we even need to get into the details of the configuration resources, just mentioning the change from using the oc adm command (which i realize is also mentioned in the deprecated features section above).

Reply:
(we can also get some input from @ahardin-rh  about what's intended to be in these notes vs what does not belong)

Reply:
It's also worth calling out because there are customers out there who install OpenShift without the integrated registry, and who would consider removing it via the operator. @bparees will this block upgrades?

Reply:
to remove it they'd mark the CR as removed.  that should not block an upgrade i do not think.

having it marked "unmanaged" might block an upgrade (but only if the registry operator propagated a "upgradeable=false" condition onto the cluster operator, i think).